cpu/cortexm_common/periph/pm.c

```diff
@@ -24,14 +24,21 @@
 #include "cpu.h"
 #include "periph/pm.h"
 
-#ifndef FEATURE_PERIPH_PM
+#ifdef NEEDS_CORTEXM_PM_FALLBACK
+#define NEEDS_CORTEXM_PM_SET_LOWEST
+#define NEEDS_CORTEXM_PM_REBOOT
+#endif
+
+#if defined(NEEDS_CORTEXM_PM_SET_LOWEST)
 void pm_set_lowest(void)
 {
     cortexm_sleep(0);
 }
 #endif
 
+#if defined(NEEDS_CORTEXM_PM_REBOOT)
```

Comment:
Question: why is `pm_reboot` guarded, if it is always used anyway? There is no case in the source tree, where the `pm_reboot` function provided by the architecture is not used... So isn't this a case of 'add the guard once we encounter a case where its needed' instead of 'adding it now because sometime in the future it might be needed'?

Reply:
Because one could define their own implementation. For all *_common CPUs I added guards around their implementations.